packages/main/src/DateRangePicker.js

```diff
@@ -0,0 +1,402 @@
+	}
+
+	_handleInputChange() {
+		const nextValue = this._getInput().getInputValue(),
+			emptyValue = nextValue === "",
+			isValid = emptyValue || this._checkValueValidity(nextValue);
+
+		if (isValid) {
+			this._setValue(nextValue);
+			this.valueState = ValueState.None;
+		} else {
+			this.valueState = ValueState.Error;
+		}
+
+		this.fireEvent("change", { value: nextValue, valid: isValid });
+		// Angular two way data binding
+		this.fireEvent("value-changed", { value: nextValue, valid: isValid });
+	}
+
+	_checkValueValidity(value) {
+		return this.isValid(value) && this.isInValidRange(value);
+	}
+
+	isValid(value) {
+		const dateStrings = this._splitValueByDelimiter(value, this.delimiter),
+			isFirstDateValid = super.isValid(dateStrings[0]),
+			isLastDateValid = super.isValid(dateStrings[1]);
+
+		if (!dateStrings[1]) {
+			return isFirstDateValid;
+		}
+
+		return isFirstDateValid && isLastDateValid;
+	}
+
+	isInValidRange(value) {
+		const dateStrings = this._splitValueByDelimiter(value, this.delimiter),
+			isFirstDateInValidRange = super.isInValidRange(this._getTimeStampFromString(dateStrings[0])),
+			isLastDateInValidRange = super.isInValidRange(this._getTimeStampFromString(dateStrings[1]));
+
+		if (!dateStrings[1]) {
+			return isFirstDateInValidRange;
+		}
+
+		return isFirstDateInValidRange && isLastDateInValidRange;
+	}
+
+	dateIntervalArrayBuilder(firstTimestamp, lastTimestamp) {
+		const datesTimestamps = [],
+			jsDate = new Date(firstTimestamp),
+			tempCalendarDate = new CalendarDate(jsDate.getFullYear(), jsDate.getMonth(), jsDate.getDate());
+
+		while (tempCalendarDate.valueOf() < lastTimestamp) {
+			datesTimestamps.push(tempCalendarDate.valueOf() / 1000);
+			tempCalendarDate.setDate(tempCalendarDate.getDate() + 1);
+		}
+
+		datesTimestamps.push(tempCalendarDate.valueOf() / 1000);
+
+		return datesTimestamps;
+	}
+
+	_handleCalendarChange(event) {
+		const newValue = event.detail.dates && event.detail.dates[0];
+		const calendarSelectedDates = this._calendar.selectedDates;
+
+		if (calendarSelectedDates[0] === newValue || calendarSelectedDates[calendarSelectedDates.length - 1] === newValue) {
+			this.closePicker();
+			return false;
+		}
+
+		if (this.isFirstDatePick) {
+			this.isFirstDatePick = false;
+			this._firstDateTimestamp = newValue;
+			this._lastDateTimestamp = newValue;
+			this._calendar.timestamp = newValue;
+		} else {
+			this.closePicker();
+			this.isFirstDatePick = true;
+			if (newValue < this._firstDateTimestamp) {
+				this._lastDateTimestamp = this._firstDateTimestamp;
+				this._firstDateTimestamp = newValue;
+			} else {
+				this._lastDateTimestamp = newValue;
+			}
+		}
+
+		const fireChange = this._handleCalendarSelectedDatesChange();
+
+		if (fireChange) {
+			this.fireEvent("change", { value: this.value, valid: true });
+			// Angular two way data binding
+			this.fireEvent("value-changed", { value: this.value, valid: true });
+		}
+	}
+
+	_handleCalendarSelectedDatesChange() {
```

Comment:
I think this method can be removed as well, the only difference with the DatePicker is what isInValidRange checks and you already have overwritten the method on line 285.

Reply:
There are differences at lines 348, 350, 351,  and the one with isInValidRange